sdk/signalr/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/core/rest/_rest_py3.py

```diff
@@ -0,0 +1,716 @@
+
+    def _close_stream(self) -> None:
+        self.is_stream_consumed = True
+        self.close()
+
+    def iter_raw(self, chunk_size: int = None) -> Iterator[bytes]:
+        """Iterate over the raw response bytes
+        """
+        self._validate_streaming_access()
+        stream_download = self._internal_response.stream_download(None)
+        for raw_bytes in stream_download:
+            self._num_bytes_downloaded += len(raw_bytes)
+            yield raw_bytes
+
+        self._close_stream()
+
+
+class AsyncHttpResponse(_HttpResponseBase):
+
+    @property
+    def content(self) -> bytes:
+        try:
+            return self._content
+        except AttributeError:
+            raise ResponseNotReadError()
+
+    async def _close_stream(self) -> None:
+        self.is_stream_consumed = True
+        await self.close()
+
+    async def read(self) -> bytes:
+        """
+        Read the response's bytes.
+
+        """
+        try:
+            return self._content
+        except AttributeError:
+            self._validate_streaming_access()
+            await self._internal_response.load_body()
+            self._content = self._internal_response._body
+            await self._close_stream()
+            return self._content
+
+    async def iter_bytes(self, chunk_size: int = None) -> Iterator[bytes]:
+        """Iterate over the bytes in the response stream
+        """
+        try:
+            chunk_size = len(self._content) if chunk_size is None else chunk_size
+            for i in range(0, len(self._content), chunk_size):
+                yield self._content[i: i + chunk_size]
+
+        except AttributeError:
+            async for raw_bytes in self.iter_raw(chunk_size=chunk_size):
+                yield raw_bytes
+
+    async def iter_text(self, chunk_size: int = None) -> Iterator[str]:
+        """Iterate over the response text
+        """
+        async for byte in self.iter_bytes(chunk_size):
+            text = byte.decode(self.encoding or "utf-8")
+            yield text
+
+    async def iter_lines(self, chunk_size: int = None) -> Iterator[str]:
+        async for text in self.iter_text(chunk_size):
+            lines = _parse_lines_from_text(text)
+            for line in lines:
+                yield line
+
+    async def iter_raw(self, chunk_size: int = None) -> Iterator[bytes]:
+        """Iterate over the raw response bytes
+        """
+        self._validate_streaming_access()
+        stream_download = self._internal_response.stream_download(None)
+        async for raw_bytes in stream_download:
+            self._num_bytes_downloaded += len(raw_bytes)
+            yield raw_bytes
+
+        await self._close_stream()
+
+    async def close(self) -> None:
+        self.is_closed = True
+        self._internal_response.internal_response.close()
+        await asyncio.sleep(0)
+
+    async def __aexit__(self, *args) -> None:
+        await self._internal_response.internal_response.__aexit__(*args)
```

Comment:
Should we also be calling `self.close()` here?

Reply:
that's a good point, let me call close instead to handle stream closing. Not sure if we want to update this in this pr @johanste , but will udpate in my core pr